### DISCUSSION TITLE
使用 LLVM19 RC，如何配置支持 import std;

### DISCUSSION THREAD

**Discussion by ebaecff** [xmake-io/xmake#5546] [Q&A]
使用VS的配置成功了，使用LLVM一直没成功，参考这个文章改的也不行：https://www.kitware.com/import-std-in-cmake-3-30/

xmake.lua
`
add_rules("mode.debug", "mode.release")

toolchain("llvm19rc")
    set_kind("standalone")
    set_sdkdir("D:/Dev/LLVM/19.1.0-rc2")
    set_bindir("D:/Dev/LLVM/19.1.0-rc2/bin")
toolchain_end()
 
set_languages("c++latest")

target("data")
    set_kind("binary")
    add_files("src/*.cpp")
    set_toolchains("llvm19rc")
    set_policy("build.c++.modules", true)
`

**Comment by waruqi** [comment]
https://github.com/xmake-io/xmake/tree/dev/tests/projects/c%2B%2B/modules/stdmodules
https://github.com/xmake-io/xmake/tree/dev/tests/projects/c%2B%2B/modules/stdmodules_cpp_only

> set_toolchains("llvm19rc")

没这个工具链，不要瞎写，只有 llvm 


```console
xmake f --toolchain=llvm --sdk=/xxxx/llvm -c
```

or 

https://github.com/xmake-io/xmake/blob/f11f39bb069741340938f1e4fa9a977fcea27348/tests/projects/package/toolchain_llvm/xmake.lua#L2-L7

**Reply by ebaecff** [reply]
照这个配置，VS可以，但 xmake f --toolchain=llvm --sdk=/xxxx/llvm -c  这个命令可以指定llvm成功，xmake模板 c++ console项目可以编译运行，但不支持 import std; 这种方式导入C++23基础模块，算了不折腾C++23了
